src/Talis/Persona/Client/Base.php

```diff
@@ -39,17 +83,21 @@
      * @param array $config An array of options with the following keys: <pre>
      *      persona_host: (string) the persona host you'll be making requests to (e.g. 'http://localhost')
      *      persona_oauth_route: (string) the token api route to query ( e.g: '/oauth/tokens')
-     *      tokencache_redis_host: (string) the host address of redis token cache
-     *      tokencache_redis_port: (integer) the port number the redis host ist listening
-     *      tokencache_redis_db: (integer) the database to connnect to</pre>
      * @param logger the logger to use, otherwise a default will be assigned and used
      * @throws \InvalidArgumentException if any of the required config parameters are missing
      */
-    public function __construct($config,\Psr\Log\LoggerInterface $logger=null) {
+    public function __construct($config, \Psr\Log\LoggerInterface $logger=null, CacheProvider $cacheBackend=null)
```

Comment:
Should the user set the `CacheProvider` through the constructor `new Login($config, $logger, $cacheBackend)` or through the setter method `$login->setCacheBackend($cacheBackend)`? As the logger and the cacheBackend are optional, it feels suitable to use a method instead.